File: 283-move-zeroes/move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let i = 0;
    let j = 0;
    while(j < nums.length) {
        if(nums[j] !== 0) {
            [nums[j], nums[i]] = [nums[i], nums[j]];
            i += 1;
        }
        j += 1;
    }
    return nums;
};

File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        if(nums.length == 1) return;
        for(int i = 0; i < nums.length-1; i++) {
            if(nums[i] == 0) {
                int j = i + 1;
                while(j < nums.length && nums[j] == 0) j += 1;
                if(i < nums.length && j < nums.length) swap(nums, i, j);
            }
        }
    }

    private void swap(int[] nums, int i, int j) {
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}